cmd/govim: investigate goimports test flake

Investigate a goimports test flake seen in #418 (unrelated to PR):

```
        --- FAIL: TestScripts/scripts/goimports (2.36s)
            testscript.go:308: 
                # Test that govim#config#Set("FormatOnSave", "goimports") and GOVIMGoImports work
                # Notice below: file.go and file.go.orig are identical. This is so that we
                # don't need to copy file.go.orig to file.go before opening file.go Because
                # doing so triggers govim to consider the file change from a file watcher
                # perspective which throws the observed diagnostics etc.
                # Ensure the default is goimports (0.031s)
                # :GOVIMGoImports whole file (1.282s)
                # Format on save (0.220s)
                # Format on save (bad syntax) (0.085s)
                > cp file.go.bad file.go
                > vim ex 'e! file.go'
                > errlogmatch -wait 30s 'sendJSONMsg: \[0,\[\d+,"call","setqflist".*'
                > vim ex 'w'
                > cmp file.go file.go.bad
                > vim expr 'getqflist()'
                [stdout]
                []
                > stdout '^\Q[{"bufnr":1,"col":1,"lnum":3,"module":"","nr":0,"pattern":"","text":"expected declaration, found blah","type":"","valid":1,"vcol":0}]\E$'
                FAIL: testdata/goimports.txt:38: no match for `^\Q[{"bufnr":1,"col":1,"lnum":3,"module":"","nr":0,"pattern":"","text":"expected declaration, found blah","type":"","valid":1,"vcol":0}]\E$` found in stdout
```

Full build log

https://gist.github.com/myitcv/f156171deab1fb891119ae4e9c707b08

Indeed, here is a failure:
```
--- FAIL: TestScripts (6.01s)
    --- FAIL: TestScripts/scripts (0.00s)
        --- FAIL: TestScripts/scripts/goimports (1.25s)
            testscript.go:308: 
                # Test that govim#config#Set("FormatOnSave", "goimports") and GOVIMGoImports work
                # Notice below: file.go and file.go.orig are identical. This is so that we
                # don't need to copy file.go.orig to file.go before opening file.go Because
                # doing so triggers govim to consider the file change from a file watcher
                # perspective which throws the observed diagnostics etc.
                # Ensure the default is goimports (0.019s)
                # :GOVIMGoImports whole file (0.404s)
                # Format on save (0.160s)
                # Format on save (bad syntax) (0.074s)
                > cp file.go.bad file.go
                > vim ex 'e! file.go'
                > errlogmatch -wait 30s 'sendJSONMsg: \[0,\[\d+,"call","setqflist".*'
                > vim ex 'w'
                > cmp file.go file.go.bad
                > vim expr 'getqflist()'
                [stdout]
                []
                > stdout '^\Q[{"bufnr":1,"col":1,"lnum":3,"module":"","nr":0,"pattern":"","text":"expected declaration, found blah","type":"","valid":1,"vcol":0}]\E$'
                FAIL: testdata/goimports.txt:38: no match for `^\Q[{"bufnr":1,"col":1,"lnum":3,"module":"","nr":0,"pattern":"","text":"expected declaration, found blah","type":"","valid":1,"vcol":0}]\E$` found in stdout
                
FAIL
```

And the corresponding log:
```
2019-08-12T21:20:13.818377_uT5AgS: vim start =======================
2019-08-12T21:20:13.818377_uT5AgS: sendJSONMsg: [0,[76,"ex","e! file.go"]]
2019-08-12T21:20:13.818377_uT5AgS: vim end =======================
2019-08-12T21:20:13.827540_uT5AgS: vim start =======================
2019-08-12T21:20:13.827540_uT5AgS: recvJSONMsg: [82] ["function","autocommand:0",[{"Num":1,"Name":"/tmp/go-test-script766403169/script-goimports/file.go","Contents":"package blah\n\nblah\n"}]]
2019-08-12T21:20:13.827540_uT5AgS: vim end =======================
2019-08-12T21:20:13.827586_uT5AgS: run: waiting to read a JSON message
2019-08-12T21:20:13.828163_uT5AgS: gopls server start =======================
2019-08-12T21:20:13.828163_uT5AgS: gopls.DidChange() call; params:
2019-08-12T21:20:13.828163_uT5AgS: &protocol.DidChangeTextDocumentParams{
2019-08-12T21:20:13.828163_uT5AgS:     TextDocument: protocol.VersionedTextDocumentIdentifier{
2019-08-12T21:20:13.828163_uT5AgS:         Version:                4,
2019-08-12T21:20:13.828163_uT5AgS:         TextDocumentIdentifier: protocol.TextDocumentIdentifier{URI:"file:///tmp/go-test-script766403169/script-goimports/file.go"},
2019-08-12T21:20:13.828163_uT5AgS:     },
2019-08-12T21:20:13.828163_uT5AgS:     ContentChanges: {
2019-08-12T21:20:13.828163_uT5AgS:         {
2019-08-12T21:20:13.828163_uT5AgS:             Range:       (*protocol.Range)(nil),
2019-08-12T21:20:13.828163_uT5AgS:             RangeLength: 0,
2019-08-12T21:20:13.828163_uT5AgS:             Text:        "package blah\n\nblah\n",
2019-08-12T21:20:13.828163_uT5AgS:         },
2019-08-12T21:20:13.828163_uT5AgS:     },
2019-08-12T21:20:13.828163_uT5AgS: }
2019-08-12T21:20:13.828163_uT5AgS: gopls server end =======================
2019-08-12T21:20:13.828212_uT5AgS: gopls server start =======================
2019-08-12T21:20:13.828212_uT5AgS: gopls.DidChange() return; err: <nil>
2019-08-12T21:20:13.828212_uT5AgS: gopls server end =======================
2019-08-12T21:20:13.828222_uT5AgS: vim start =======================
2019-08-12T21:20:13.828222_uT5AgS: sendJSONMsg: [82,["",null]]
2019-08-12T21:20:13.828222_uT5AgS: vim end =======================
2019-08-12T21:20:13.830945_uT5AgS: vim start =======================
2019-08-12T21:20:13.830945_uT5AgS: recvJSONMsg: [83] ["callback",76,[""]]
2019-08-12T21:20:13.830945_uT5AgS: vim end =======================
2019-08-12T21:20:13.830969_uT5AgS: run: waiting to read a JSON message
2019-08-12T21:20:13.834488_uT5AgS: vim start =======================
2019-08-12T21:20:13.834488_uT5AgS: sendJSONMsg: [0,[77,"ex","w"]]
```

Note that there is no `setqflist` call between the second and the last line (but there is earlier in the log that isn't included here).

A solution could be to add `-count` & `-start` to the specific `errlogmatch` to ensure that it is the intended match that is found (or make sure that the old log isn't searched by calling `errlogmatch` before)

--- a/cmd/govim/testdata/goimports.txt
+++ b/cmd/govim/testdata/goimports.txt
@@ -31,7 +31,7 @@ errlogmatch -count=0 'LogMessage callback: &protocol\.LogMessageParams\{Type:%v,
 # Format on save (bad syntax)
 cp file.go.bad file.go
 vim ex 'e! file.go'
-errlogmatch -wait 30s 'sendJSONMsg: \[0,\[\d+,"call","setqflist".*'
+errlogmatch -wait 30s -start -count=5 'sendJSONMsg: \[0,\[\d+,"call","setqflist".*'
 vim ex 'w'
 cmp file.go file.go.bad
 vim expr 'getqflist()'

